File: Microservices/Authentication-Service/src/main/java/com/xws/tim12/model/Agent.java
package com.xws.tim12.model;

import java.util.Collection;
import java.util.Set;

import javax.persistence.CascadeType;
import javax.persistence.Column;
import javax.persistence.Entity;
import javax.persistence.EnumType;
import javax.persistence.Enumerated;
import javax.persistence.FetchType;
import javax.persistence.GeneratedValue;
import javax.persistence.GenerationType;
import javax.persistence.Id;
import javax.persistence.JoinColumn;
import javax.persistence.JoinTable;
import javax.persistence.ManyToMany;
import javax.validation.constraints.Email;
import javax.validation.constraints.NotEmpty;
import javax.validation.constraints.NotNull;

import org.springframework.security.core.GrantedAuthority;
import org.springframework.security.core.userdetails.UserDetails;

import com.xws.tim12.enumeration.EntityType;

import net.minidev.json.annotate.JsonIgnore;

@Entity
public class Agent implements UserDetails {

    @Id
    @Column
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @NotEmpty(message = "Username cannot be empty.")
    @NotNull(message = "Username cannot be null.")
    @Column(nullable = false)
    private String username;

    @JsonIgnore
    @NotEmpty(message = "Password cannot be empty.")
    @NotNull(message = "Password cannot be null.")
    @Column(nullable = false)
    private String password;

    @Column(columnDefinition = "VARCHAR(30)", nullable = true)
    private String firstName;

    @Column(columnDefinition = "VARCHAR(30)", nullable = true)
    private String lastName;

    @Email()
    private String email;

    @Enumerated(EnumType.STRING)
    private EntityType entityType;

    @Column(nullable = true)
    private String companyName;

    @Column(nullable = true)
    private String businessRegistrationNumber;

    @Column(nullable = true)
    private String address;

    @Column()
    private Boolean isBanned;

    @ManyToMany(cascade = CascadeType.ALL, fetch = FetchType.EAGER)
    @JoinTable(
            name = "agent_authority",
            joinColumns = @JoinColumn(name = "agent_id", referencedColumnName = "id"),
            inverseJoinColumns = @JoinColumn(name = "authority_id", referencedColumnName = "id"))
    private Set<Authority> authorities;

    public Agent() {

    }

    public Agent(
            @NotEmpty(message = "Username cannot be empty.")
            @NotNull(message = "Username cannot be null.") String username,
            @NotEmpty(message = "Password cannot be empty.")
            @NotNull(message = "Password cannot be null.") String password,
            String firstName,
            String lastName,
            @Email() String email,
            EntityType entityType,
            String companyName,
            @NotEmpty(message = "Business registration number cannot be empty.") String businessRegistrationNumber,
            @NotEmpty(message = "Address cannot be empty.") String address,
            Boolean isBanned) {
        this.username = username;
        this.password = password;
        this.firstName = firstName;
        this.lastName = lastName;
        this.email = email;
        this.entityType = entityType;
        this.companyName = companyName;
        this.businessRegistrationNumber = businessRegistrationNumber;
        this.address = address;
        this.isBanned = isBanned;
    }

    public Long getId() {
        return id;
    }

    public void setId(Long id) {
        this.id = id;
    }

    public String getUsername() {
        return username;
    }

    @Override
    public boolean isAccountNonExpired() {
        return true;
    }

    @Override
    public boolean isAccountNonLocked() {
        return true;
    }

    @Override
    public boolean isCredentialsNonExpired() {
        return true;
    }

    @Override
    public boolean isEnabled() {
        return true;
    }

    public void setUsername(String username) {
        this.username = username;
    }

    @Override
    public Collection<? extends GrantedAuthority> getAuthorities() {
        return this.authorities;
    }

    public String getPassword() {
        return password;
    }

    public void setPassword(String password) {
        this.password = password;
    }

    public String getFirstName() {
        return firstName;
    }

    public void setFirstName(String firstName) {
        this.firstName = firstName;
    }

    public String getLastName() {
        return lastName;
    }

    public void setLastName(String lastName) {
        this.lastName = lastName;
    }

    public String getEmail() {
        return email;
    }

    public void setEmail(String email) {
        this.email = email;
    }

    public EntityType getEntityType() {
        return entityType;
    }

    public void setEntityType(EntityType entityType) {
        this.entityType = entityType;
    }

    public String getCompanyName() {
        return companyName;
    }

    public void setCompanyName(String companyName) {
        this.companyName = companyName;
    }

    public String getBusinessRegistrationNumber() {
        return businessRegistrationNumber;
    }

    public void setBusinessRegistrationNumber(String businessRegistrationNumber) {
        this.businessRegistrationNumber = businessRegistrationNumber;
    }

    public String getAddress() {
        return address;
    }

    public void setAddress(String address) {
        this.address = address;
    }

    public Boolean getBanned() {
        return isBanned;
    }

    public void setBanned(Boolean banned) {
        isBanned = banned;
    }

    @Override
    public String toString() {
        return "Agent{" +
                "id=" + id +
                ", username='" + username + '\'' +
                ", password='" + password + '\'' +
                ", firstName='" + firstName + '\'' +
                ", lastName='" + lastName + '\'' +
                ", email='" + email + '\'' +
                ", entityType=" + entityType +
                ", companyName='" + companyName + '\'' +
                ", businessRegistrationNumber='" + businessRegistrationNumber + '\'' +
                ", address='" + address + '\'' +
                ", isBanned=" + isBanned +
                '}';
    }

}

File: Frontend/src/app/models/agent.ts
export class Agent {

	private id: number;
	private username: string;
	private password: string;
	private firstName: string;
	private lastName: string;
	private email: string;
	private companyName: string;
	private businessRegistrationNumber: string;
	private address: string;
	private isBanned: boolean;

	constructor($id: number, $username: string, $password: string, $firstName: string, $lastName: string, $email: string, $companyName: string, $businessRegistrationNumber: string, $address: string, $isBanned: boolean) {
		this.id = $id;
		this.username = $username;
		this.password = $password;
		this.firstName = $firstName;
		this.lastName = $lastName;
		this.email = $email;
		this.companyName = $companyName;
		this.businessRegistrationNumber = $businessRegistrationNumber;
		this.address = $address;
		this.isBanned = $isBanned;
	}

    /**
     * Getter $id
     * @return {number}
     */
	public get $id(): number {
		return this.id;
	}

    /**
     * Getter $username
     * @return {string}
     */
	public get $username(): string {
		return this.username;
	}

    /**
     * Getter $password
     * @return {string}
     */
	public get $password(): string {
		return this.password;
	}

    /**
     * Getter $firstName
     * @return {string}
     */
	public get $firstName(): string {
		return this.firstName;
	}

    /**
     * Getter $lastName
     * @return {string}
     */
	public get $lastName(): string {
		return this.lastName;
	}

    /**
     * Getter $email
     * @return {string}
     */
	public get $email(): string {
		return this.email;
	}

    /**
     * Getter $companyName
     * @return {string}
     */
	public get $companyName(): string {
		return this.companyName;
	}

    /**
     * Getter $businessRegistrationNumber
     * @return {string}
     */
	public get $businessRegistrationNumber(): string {
		return this.businessRegistrationNumber;
	}

    /**
     * Getter $address
     * @return {string}
     */
	public get $address(): string {
		return this.address;
	}

    /**
     * Getter $isBanned
     * @return {boolean}
     */
	public get $isBanned(): boolean {
		return this.isBanned;
	}

    /**
     * Setter $id
     * @param {number} value
     */
	public set $id(value: number) {
		this.id = value;
	}

    /**
     * Setter $username
     * @param {string} value
     */
	public set $username(value: string) {
		this.username = value;
	}

    /**
     * Setter $password
     * @param {string} value
     */
	public set $password(value: string) {
		this.password = value;
	}

    /**
     * Setter $firstName
     * @param {string} value
     */
	public set $firstName(value: string) {
		this.firstName = value;
	}

    /**
     * Setter $lastName
     * @param {string} value
     */
	public set $lastName(value: string) {
		this.lastName = value;
	}

    /**
     * Setter $email
     * @param {string} value
     */
	public set $email(value: string) {
		this.email = value;
	}

    /**
     * Setter $companyName
     * @param {string} value
     */
	public set $companyName(value: string) {
		this.companyName = value;
	}

    /**
     * Setter $businessRegistrationNumber
     * @param {string} value
     */
	public set $businessRegistrationNumber(value: string) {
		this.businessRegistrationNumber = value;
	}

    /**
     * Setter $address
     * @param {string} value
     */
	public set $address(value: string) {
		this.address = value;
	}

    /**
     * Setter $isBanned
     * @param {boolean} value
     */
	public set $isBanned(value: boolean) {
		this.isBanned = value;
	}

}